packages/appium-tizen-tv-driver/lib/driver.js

```diff
@@ -180,43 +311,110 @@ class TizenTVDriver extends BaseDriver {
       try {
         await this.chromedriver.stop();
       } catch (err) {
-        log.warn(`Error stopping Chromedriver: ${err.message}`);
+        log.warn(`Error stopping Chromedriver: ${/** @type {Error} */ (err).message}`);
       }
       this.chromedriver = null;
     }
 
-    await this.remote.disconnect();
-    this.remote = null;
+    if (this.#remote) {
+      await this.#remote.disconnect();
+      this.#remote = undefined;
+    }
     await this.cleanUpPorts();
     return await super.deleteSession();
   }
 
   async cleanUpPorts() {
     log.info(`Cleaning up any ports which have been forwarded`);
-    for (const localPort of this.forwardedPorts) {
-      await removeForwardedPort({udid: this.opts.udid, localPort});
+    for (const localPort of this.#forwardedPorts) {
+      await removeForwardedPort({udid: /** @type {string} */(this.opts.udid), localPort});
     }
   }
 
   proxyActive() {
-    return this.jwpProxyActive;
+    return this.#jwpProxyActive;
   }
 
   getProxyAvoidList() {
-    return this.jwpProxyAvoid;
+    return this.#jwpProxyAvoid;
   }
 
   canProxy() {
     return true;
   }
 
-  async pressKeyCode(keycode) {
-    if (!Keys[keycode]) {
-      throw new Error(`Keycode '${keycode}' was not recognized`);
+  /**
+   *
+   * @param {RcKeyCode} rcKeyCode
+   */
+  async pressKey(rcKeyCode) {
+    if (!isRcKeyCode(rcKeyCode)) {
+      throw new TypeError(`Invalid key code: ${rcKeyCode}`);
     }
-    await this.remote.click(keycode);
+    if (this.#remote) {
```

Comment:
should we not check for `this.opts.rcMode` instead?

Reply:
if we're being defensive, we should check for both, since `this.#remote` should not be present if `rcMode` is `js`.  at least, that's the intent.  if either `this.#remote` _or_ `rcMode == 'js'` is truthy, but not both, then we should probably throw, because that's a developer bug.

a better way to do this would be a subclass that has a `this.#remote`, perhaps, which is created when `rcMode` is not `js`.  given appium's "one class per extension" requirement, this would demand the "main class" to be a factory function which returns an instance of the proper class.  anyhow, if it becomes problematic later, maybe we should refactor it.

Reply:
> if either this.#remote or rcMode == 'js' is truthy, but not both, then we should probably throw, because that's a developer bug.

for now i think this is the way to go, just to make sure a future dev doesn't inadvertently change how this works without realizing we're relying on this equivalence.